signer/README.md

```diff
@@ -0,0 +1,82 @@
+# Mina signer
+
+This crate provides an API and framework for Mina signing.  It follows the algorithm outlined in the [Mina Signature Specification](https://github.com/MinaProtocol/mina/blob/develop/docs/specs/signatures/description.md).
+
+## Simple interface
+
+The simple interface uses the default signer configuration compatible with mainnet and testnet transaction signatures.
+
+```rust
+use rand;
+use mina_signer::{NetworkId, Keypair, Signer};
+use mina_signer::NetworkId;
+
+let mut ctx = mina_signer::create(NetworkId::TESTNET);
+let sig = ctx.sign(key_pair, transaction);
+
+assert_eq!(ctx.verify(sig, key_pair.public, transaction), true);
+```
+
+## Advanced interface
+
+The advanced interface allows specification of an alternative cryptographic sponge and parameters, for example, in order to create signatures that can be verified more efficiently using the Kimchi proof system.
+
+```rust
+use rand;
+use mina_signer::{NetworkId, Keypair, Signer};
+use oracle::{pasta, poseidon};
+
+let mut ctx = mina_signer::custom::<poseidon::PlonkSpongeConstants5W>(
+    pasta::fp5::params(),
```

Comment:
why is poseidon stuff being passed twice, once as generic param and once as argument?

Reply:
This is just from how proof-systems sponge works.  They are two different things.  The generic param is the `SpongeConstants`, which configures the sponge capacity, rate, rounds, s-box alpha, width and a couple of other things.  The argument is the `ArithmeticSpongeParams` which just contains the round constants and MDS matrix.  If the sponge, it's consts and params are ever refactored into a single parameter then we can change this API to just use the single thing.  For now, unfortunately they come as two things from two different places.